browser/ui/toolbar/brave_app_menu_model.cc

```diff
@@ -31,62 +17,144 @@ void BraveAppMenuModel::Build() {
 }
 
 void BraveAppMenuModel::InsertBraveMenuItems() {
-  // Sync & Rewards pages are redirected to normal window when it is loaded in
-  // private window. So, only hide them in guest(tor) window.
-  if (!browser_->profile()->IsGuestSession()) {
-    bool walletEnabled = browser_->profile()->GetPrefs()->
-        GetBoolean(kBraveWalletEnabled);
-    InsertItemWithStringIdAt(
-        GetIndexOfCommandId(IDC_SHOW_DOWNLOADS),
-        IDC_SHOW_BRAVE_REWARDS,
-        IDS_SHOW_BRAVE_REWARDS);
-    if (walletEnabled) {
-#if BUILDFLAG(BRAVE_WALLET_ENABLED)
-      InsertItemWithStringIdAt(GetIndexOfCommandId(IDC_SHOW_BRAVE_REWARDS),
-                               IDC_SHOW_BRAVE_WALLET, IDS_SHOW_BRAVE_WALLET);
-#endif
-    }
-    InsertItemWithStringIdAt(
-#if BUILDFLAG(BRAVE_WALLET_ENABLED)
-        GetIndexOfCommandId(walletEnabled ? IDC_SHOW_BRAVE_WALLET :
-            IDC_SHOW_BRAVE_REWARDS),
-#else
-        GetIndexOfCommandId(IDC_SHOW_BRAVE_REWARDS),
-#endif
-        IDC_SHOW_BRAVE_SYNC, IDS_SHOW_BRAVE_SYNC);
-  }
+  // Insert & reorder brave menus based on corresponding commands enable status.
+  // If we you want to add/remove from app menu, adjust commands enable status
+  // at BraveBrowserCommandController.
 
-  // Insert Create New Profile item
-  auto profile_items_insert_index = GetIndexOfCommandId(IDC_ZOOM_MENU) - 1;
-  InsertItemWithStringIdAt(profile_items_insert_index,
-    IDC_OPEN_GUEST_PROFILE,
-    IDS_OPEN_GUEST_PROFILE);
-  InsertItemWithStringIdAt(
-    GetIndexOfCommandId(IDC_OPEN_GUEST_PROFILE),
-    IDC_ADD_NEW_PROFILE,
-    IDS_ADD_NEW_PROFILE);
-  InsertSeparatorAt(
-    GetIndexOfCommandId(IDC_ADD_NEW_PROFILE), ui::NORMAL_SEPARATOR);
-
-  InsertItemWithStringIdAt(
-      GetIndexOfCommandId(IDC_SHOW_DOWNLOADS),
-      IDC_SHOW_BRAVE_ADBLOCK,
-      IDS_SHOW_BRAVE_ADBLOCK);
-  if (brave::IsTorProfile(browser_->profile())) {
+  // Step 1. Configure tab & windows section.
+  if (IsCommandIdEnabled(IDC_NEW_TOR_CONNECTION_FOR_SITE)) {
     InsertItemWithStringIdAt(
         GetIndexOfCommandId(IDC_NEW_WINDOW),
         IDC_NEW_TOR_CONNECTION_FOR_SITE,
         IDS_NEW_TOR_CONNECTION_FOR_SITE);
   }
-
   if (IsCommandIdEnabled(IDC_NEW_OFFTHERECORD_WINDOW_TOR)) {
     InsertItemWithStringIdAt(GetIndexOfCommandId(IDC_NEW_INCOGNITO_WINDOW) + 1,
                              IDC_NEW_OFFTHERECORD_WINDOW_TOR,
                              IDS_NEW_OFFTHERECORD_WINDOW_TOR);
   }
 
-  InsertItemWithStringIdAt(
-      GetIndexOfCommandId(IDC_ABOUT),
-      IDC_SHOW_BRAVE_WEBCOMPAT_REPORTER,
-      IDS_SHOW_BRAVE_WEBCOMPAT_REPORTER);
+  // Step 2. Configure second section that includes history, downloads and
+  // bookmark. Then, insert brave items.
+
+  // First, reorder original menus We want to move them in order of bookmark,
+  // download and extensions.
+  const int bookmark_item_index = GetIndexOfCommandId(IDC_BOOKMARKS_MENU);
+  // If bookmark is not used, we don't need to adjust download item.
```

Comment:
When will bookmark menu item become unavailable?

Reply:
guest profile disables it.